packages/story-editor/src/app/media/utils/useMediaUploadQueue/index.js

```diff
@@ -423,111 +425,142 @@ function useMediaUploadQueue() {
     [startUploading, cancelUploading, uploadImage, uploadVideo]
   );
 
-  // Upload files to server, optionally first transcoding them.
-  useEffect(() => {
-    (async () => {
-      await Promise.all(
-        /**
-         * Uploads a single pending item.
-         *
-         * @param {Object} item Queue item.
-         * @param {File} item.file File object.
-         * @return {Promise<void>}
-         */
-        state.queue.map(async (item) => {
-          const {
-            id,
-            file,
-            state: itemState,
-            resource,
-            additionalData = {},
-            muteVideo,
-            trimData,
-          } = item;
-          if (ITEM_STATUS.PENDING !== itemState) {
+  const uploadWithTranscoding = useCallback(
+    async (item) => {
+      const { id, file, resource, muteVideo, trimData } = item;
+
+      const isVideo = file.type.startsWith('video/');
+
+      if (isVideo) {
+        const fileInfo = await getFileInfo(file);
```

Comment:
Why do we need to do a call to `getFileInfo` here?  I did some testing and console.logging and between file and resource objects you have all this data. See the screenshot. 
<img width="733" alt="Screenshot 2022-07-01 at 16 49 26" src="https://user-images.githubusercontent.com/237508/176928406-4db1e17f-cc00-47b1-bb55-bf6f9fdb0294.png">


Could we not just do this. 
```
const fileInfo = {
    fileSize: file.size, 
    height: resource.height,
    width: resource.width,
    mimeType: resource.mimeType,
    duration: resource.length,
};
```

If we do this, we can save an expensive call to `getFileInfo`. If resource / file is not aviable, fallback to getFileInfo. We could even do a first check with resource data, to get height/width/file size/mime type. Then check other things later.

Reply:
To answer your question(s), for the start in this PR the check is very simple since we have to nail down the right criteria first. And for that we definitely need to take codecs and frame rate into account as well. As you've seen at #11405. And we only get that information via `getFileInfo`, which is the whole purpose of this ticket.

The exact criteria will follow in a separate PR.

Reply:
Can we do two checks? One for simple data, like height, width, duration, mime type and file size. If that check passes then do the call to file info?

Reply:
You mean like if it‘s a MOV file don‘t bother getting file info since it needs transcoding anyway?

Reply:
> You mean like if it‘s a MOV file don‘t bother getting file info since it needs transcoding anyway?

Yes! Why load this library if we don't need to. Do a basic check for mime type, dimensions and length. If these fail check ( as in not mp4 etc ), then skill loading this library and just transcode. This is "preflight" check passes, then do the check with fileInfo and loading that library. Most transcoding happens, because the file type / size is wrong. We may never need to load this library.

Reply:
Yeah that makes sense.

The only downside is that we can't do this anymore:

https://github.com/GoogleForCreators/web-stories-wp/blob/cbd8e59cf3f85d59b48ac761d1bcef9ee5c46932/packages/story-editor/src/app/media/utils/useMediaUploadQueue/index.js#L438-L440

Reply:
So in `getResourceFromLocalFile` we use `hasVideoGotAudio` to determine whether a video has audio or not.

This might not work for all types of videos, if the browser cannot properly play it. Think MOV, MKV, etc. It might mark a video as muted even though it _does_ have audio.

By using `getFileInfo` here we have more accurate audio information.

Reply:
Could we not do this?

```
  if (isVideo) {
        if (isConsideredOptimizedResource(resource)) {
            const fileInfo = await getFileInfo(file);

           if (fileInfo) {
             // This more accurate audio information is also useful
            // if transcoding is not skipped, so we set this here.
            item.additionalData.isMuted = fileInfo.isMuted;

             if (isConsideredOptimized(fileInfo)) {
               // Do not override pre-existing mediaSource if provided,
               // for example by media recording.
               if (!item.additionalData.mediaSource) {
                 item.additionalData.mediaSource = 'video-optimization';
                }

                uploadItem(item);

                return;
               }
         } 
        }
      }
```

Reply:
Refactored this a bit in 8b65a5c235762b42e00e7ec61df0c89b0793aef2

At the cost of the more accurate audio information, but with short-circuit